diff --git a/layer/meson.build b/layer/meson.build
index 830cd89..86e6efb 100644
--- a/layer/meson.build
+++ b/layer/meson.build
@@ -1,6 +1,7 @@
 vkroots_dep = dependency('vkroots')
 dep_xcb = dependency('xcb')
 dep_x11_xcb = dependency('x11-xcb')
+glm_dep = dependency('glm')
 wayland_client = dependency('wayland-client')
 
 gamescope_wsi_layer = shared_library('VkLayer_FROG_gamescope_wsi_' + build_machine.cpu_family(), 'VkLayer_FROG_gamescope_wsi.cpp', protocols_client_src,
diff --git a/meson.build b/meson.build
index 9050f00..562ee15 100644
--- a/meson.build
+++ b/meson.build
@@ -50,11 +50,6 @@ dep_x11 = dependency('x11')
 dep_wayland = dependency('wayland-client')
 vulkan_dep = dependency('vulkan')
 
-glm_proj = subproject('glm')
-glm_dep = glm_proj.get_variable('glm_dep')
-stb_proj = subproject('stb')
-stb_dep = stb_proj.get_variable('stb_dep')
-
 if get_option('enable_openvr_support')
   openvr_dep = dependency('openvr', version: '>= 2.7', required : false)
   if not openvr_dep.found()
diff --git a/src/meson.build b/src/meson.build
index a3dfdab..bd25fb5 100644
--- a/src/meson.build
+++ b/src/meson.build
@@ -21,7 +21,9 @@ xkbcommon = dependency('xkbcommon')
 thread_dep = dependency('threads')
 cap_dep = dependency('libcap', required: get_option('rt_cap'))
 epoll_dep = dependency('epoll-shim', required: false)
+glm_dep = dependency('glm')
 sdl2_dep = dependency('SDL2', required: get_option('sdl2_backend'))
+stb_dep = dependency('stb')
 avif_dep = dependency('libavif', version: '>=1.0.0', required: get_option('avif_screenshots'))
 
 wlroots_dep = dependency(
