diff --git a/backend/libinput/meson.build b/backend/libinput/meson.build
index c244eb77a529c337cb99fad28e47981727d45a41..091b0e0eb00b921f1b407a5d64d9cb1bddd7f9ce 100644
--- a/subprojects/wlroots/backend/libinput/meson.build
+++ a/subprojects/wlroots/backend/libinput/meson.build
@@ -29,3 +29,7 @@ features += { 'libinput-backend': true }
 wlr_deps += libinput
 
 internal_config.set10('HAVE_LIBINPUT_BUSTYPE', libinput.version().version_compare('>=1.26.0'))
+internal_config.set10(
+	'HAVE_LIBINPUT_SWITCH_KEYPAD_SLIDE',
+	libinput.version().version_compare('>=1.30.901')
+)
diff --git a/backend/libinput/switch.c b/backend/libinput/switch.c
index abeec86d7ab3b9d0e4cb101f413666338be842fc..a78d37abea6f4d0e82080a9c0b095e96ab46351e 100644
--- a/subprojects/wlroots/backend/libinput/switch.c
+++ a/subprojects/wlroots/backend/libinput/switch.c
@@ -2,6 +2,7 @@
 #include <libinput.h>
 #include <wlr/interfaces/wlr_switch.h>
 #include "backend/libinput.h"
+#include "config.h"
 
 const struct wlr_switch_impl libinput_switch_impl = {
 	.name = "libinput-switch",
@@ -36,6 +37,11 @@ void handle_switch_toggle(struct libinput_event *event,
 	case LIBINPUT_SWITCH_TABLET_MODE:
 		wlr_event.switch_type = WLR_SWITCH_TYPE_TABLET_MODE;
 		break;
+#if HAVE_LIBINPUT_SWITCH_KEYPAD_SLIDE
+	case LIBINPUT_SWITCH_KEYPAD_SLIDE:
+		wlr_event.switch_type = WLR_SWITCH_TYPE_KEYPAD_SLIDE;
+		break;
+#endif
 	}
 	switch (libinput_event_switch_get_switch_state(sevent)) {
 	case LIBINPUT_SWITCH_STATE_OFF:
diff --git a/include/wlr/types/wlr_switch.h b/include/wlr/types/wlr_switch.h
index 641df19910510753cd1d2f1f345e02060058b119..2b9e91266e55511e5ffb9b15be8c12965593e588 100644
--- a/subprojects/wlroots/include/wlr/types/wlr_switch.h
+++ a/subprojects/wlroots/include/wlr/types/wlr_switch.h
@@ -36,6 +36,7 @@ struct wlr_switch {
 enum wlr_switch_type {
 	WLR_SWITCH_TYPE_LID,
 	WLR_SWITCH_TYPE_TABLET_MODE,
+	WLR_SWITCH_TYPE_KEYPAD_SLIDE,
 };
 
 enum wlr_switch_state {
